app/src/features/sources/sourceSlice.ts

```diff
@@ -0,0 +1,23 @@
+import { createSlice, createAction } from "@reduxjs/toolkit";
+
+import type { RootState } from "../../app/store";
+import type { Source } from "../../services/api/generated/api.generated";
+
+const initialState = null as null | Source;
```

Comment:
What do you think about adding a field `isEditingSource` to avoid checking if source exists or not?

Reply:
I can add a selector which will return a boolean